Automattic/jetpack
# VideoPress: Add check to remove tracks from previous video after replacement on block

Fixes #29464

## Proposed changes:

* Adds a check for extra tracks in the block attributes that are not in the video data when syncing tracks

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Have at least 3 videos uploaded to VideoPress, two of which should have chapter tracks and one without
* Go to the block editor
* Add a VideoPress video block with a video that has tracks
* Check that the tracks are listed
![2023-03-15_17-27-48](https://user-images.githubusercontent.com/8486249/225434430-89fb823b-8387-45c8-812c-f051a991fd07.png)
* Click on `Replace` and, through the Media Library, select the other video that has chapters
* Check that the chapters listed are correct
* Click on `Replace` again, this time selecting the video without chapter tracks
* Check that the chapter tracks list is empty - *This is where it was failing before this PR*
* Repeat the tests, now using the video `guid`s in the URL option
![2023-03-15_17-31-07](https://user-images.githubusercontent.com/8486249/225435053-9d58f92b-2641-4b15-b9af-0e54930f823f.png)
* Check that the replacements happen as normal and the chapter tracks list is updated correctly
* Replace a block that has tracks with a new one via upload
* Check that the chapter tracks list is empty
